packages/repo/tests/sync/narrow.test.ts

```diff
@@ -0,0 +1,140 @@
+import { TID } from '@atproto/common'
+import * as crypto from '@atproto/crypto'
+import { DidResolver } from '@atproto/did-resolver'
+import { RecordClaim, Repo, RepoContents } from '../../src'
+import { MemoryBlockstore } from '../../src/storage'
+import * as verify from '../../src/verify'
+import * as sync from '../../src/sync'
+
+import * as util from '../_util'
+
+describe('Narrow Sync', () => {
+  let storage: MemoryBlockstore
+  let repo: Repo
+  let keypair: crypto.Keypair
+  let repoData: RepoContents
+  const didResolver = new DidResolver()
+
+  beforeAll(async () => {
+    storage = new MemoryBlockstore()
+    keypair = await crypto.Secp256k1Keypair.create()
+    repo = await Repo.create(storage, keypair.did(), keypair)
+    const filled = await util.fillRepo(repo, keypair, 5)
+    repo = filled.repo
+    repoData = filled.data
+  })
+
+  const getProofs = async (claims: RecordClaim[]) => {
+    return sync.getRecords(storage, repo.cid, claims)
+  }
+
+  const doVerify = (proofs: Uint8Array, claims: RecordClaim[]) => {
+    return verify.verifyProofs(repo.did, proofs, claims, didResolver)
+  }
+
+  it('verifies valid records', async () => {
+    const claims = util.contentsToClaims(repoData)
+    const proofs = await getProofs(claims)
+    const results = await doVerify(proofs, claims)
+    expect(results.verified).toEqual(claims)
```

Comment:
Probably a bit of a paranoid move, but an extra check here would guarantee that some verification is definitely happening in the test:
```suggestion
    expect(claims.length).toBeGreaterThan(0)
    expect(results.verified).toEqual(claims)
```

Reply:
i dig 👌